arangod/IResearch/IResearchLink.cpp

```diff
@@ -1840,7 +2055,8 @@ Result IResearchLink::insert(transaction::Methods& trx,
     ctx = ptr.get();
     state.cookie(key, std::move(ptr));
 
-    if (!ctx || !trx.addStatusChangeCallback(&_trxCallback)) {
+    if (!ctx || !trx.addStatusChangeCallback(&_trxCallback) ||
+        !trx.addPreCommitCallback(&_trxPreCommit)) {
```

Comment:
Same here

Reply:
I guess not. As transaction will be aborted with error anyway.